integrated_channels/api/v1/degreed/views.py

```diff
@@ -0,0 +1,41 @@
+"""
+Viewsets for Degreed
+"""
+
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+
+from enterprise.models import EnterpriseCustomerUser
+from integrated_channels.degreed.models import DegreedEnterpriseCustomerConfiguration
+
+from .serializers import DegreedConfigSerializer
+
+
+class DegreedConfigurationViewSet(PermissionRequiredMixin, viewsets.ModelViewSet):
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
```

Comment:
So I ended up adding these because I couldn't actually test the API directly via postman without it. Basically by using postman I am bypassing LMS/Platform's authentication pathway and needed a way to actually do so which is what this allows.

Most of our APIs actually have these authentication classes (enterprise itself is one example). However, the actual implementation details I have not investigated so can't provide more detail atm about the exact mechanisms.

Reply:
I believe that is exactly what Postman Interceptor is for- In terms of bypassing LMS' authentication pathway. Let me know if you want any info on it, @edx-abolger is also a good reference as she showed me 🙂

Reply:
I think the use of these classes is more something we probably want to discuss with Titans since I don't know if that is the *only* use of these authentication classes given how expansively they are used in this repo. As far as the utility, yeah if I don't use these classes that will likely come in handy.

For now, leaving it or not is pretty harmless (I mean if there was a security concern then I'd be eyeing like 90% of our API endpoints in edx-enterprise, so uhhh I hope not). It has a use, but if we wanted to adopt this interceptor workflow then that use would be unnecessary.

Reply:
My uninformed guess is that they come in handy when hitting the endpoints outside of an authentication process (ie backend to backend) which is likely to be forever irrelevant for these particular APIs

Reply:
fwiw @alex-sheehan-edx there is an unused import of jwtauthentication in blackboard (sorry not related to this pr so we can take offline) https://github.com/edx/edx-enterprise/blob/de43ebeb0f147e67b8d5fbdbac19579b89c7ab8c/integrated_channels/api/v1/blackboard/views.py#L5 I cannot find usage of jwtauthetnication in any of the new channels we worked on (but they exist in other channels). we should find out more.

Reply:
my understanding is that the `PermissionRequiredMixin` handles the RBAC by only allowing users with the right roles, but it was not clear to me if it depended on the jwtAuthentication being present.

Reply:
yeah and it does not depend on there being a jwt auth of course, e.g., https://github.com/edx/edx-enterprise/blob/320ea9842175b23d40f4f32a80806e094cd85421/integrated_channels/api/v1/canvas/views.py#L18 so my guess is you don't need it

Reply:
yeah I just went ahead and removed it from my PR.